src/__tests__/downshift.misc.js

```diff
@@ -116,6 +116,22 @@ test('can use children instead of render prop', () => {
   expect(childrenSpy).toHaveBeenCalledTimes(1)
 })
 
+test('should not throw error during strict mode during reset', () => {
```

Comment:
Thanks for this! Do you mind showing me this test failing without your changes? I'm really sorry, I'm just struggling to see how your changes really impact anything here :-/

Reply:
How would you like me to do that? Would you like me to revert my change in downshift and push? I can do that first thing tomorrow when I get to work.

Reply:
A screenshot of the failing test would suffice.

Or if you could point me to the part of your code that's making the real impact or explain how this is making a difference. I'm sure I'm missing something, but it looks like this is just renaming variables...

Reply:
![image](https://user-images.githubusercontent.com/6361167/38877580-fad1ab7a-422c-11e8-812f-5cd4de12209e.png)

@kentcdodds - Here is a screenshot of the failing test.

Let me try to explain a little more of the issue.

The `internalSetState` function is not invoked twice by react when in Strict mode. The part that is invoked twice is the updater function that is passed to `this.setState`. The first time the updater function is invoked, `stateToSet` is a function (which was passed into internalSetState). Inside the updater function, the `stateToSet` variable gets reassigned to the result of that function call which is an object (the new state). React then invokes the updater function again (as part of StrictMode). This time, `stateToSet` is an object since that was changed during the previous invocation.

My change ensures that `stateToSet` variable never gets reassigned so that it remains the same on subsequent calls of the updater function.

Reply:
Here is a CodeSandbox of the issue if this helps:
[![Edit 8446ynnn7j](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/8446ynnn7j)